src/containers/pages/InterventionPlan/IRS/plan/index.tsx

```diff
@@ -1,79 +1,1419 @@
+      this.onSelectCountryChange(e);
+    };
+    const onStartPlanFormSubmit = (e: any) => {
+      this.onStartPlanFormSubmit(e);
+    };
+
+    if (isStartingPlan && newPlan) {
+      const { plan_effective_period_end, plan_effective_period_start, plan_title } = newPlan;
+      return (
+        <div className="mb-5">
+          <Helmet>
+            <title>IRS: New Plan</title>
+          </Helmet>
+          <HeaderBreadcrumbs {...breadCrumbProps} />
+          <Row>
+            <Col>
+              <h2 className="page-title">New IRS Plan</h2>
+            </Col>
+          </Row>
+          <Row>
+            <Col xs="6">
+              <Form>
+                <FormGroup>
+                  <Label for="set-plan-name">Plan Name</Label>
+                  <Input
+                    id="set-plan-name"
+                    onChange={onSetPlanNameChange}
+                    placeholder={plan_title}
+                  />
+                </FormGroup>
+                <FormGroup>
+                  <Label for="set-plan-start-date">Start Date</Label>
+                  <Input
+                    id="set-plan-start-date"
+                    onChange={onSetPlanStartDateChange}
+                    defaultValue={plan_effective_period_start}
+                    type="date"
+                  />
+                </FormGroup>
+                <FormGroup>
+                  <Label for="set-plan-end-date">End Date</Label>
+                  <Input
+                    id="set-plan-end-date"
+                    onChange={onSetPlanEndDateChange}
+                    defaultValue={plan_effective_period_end}
+                    type="date"
+                  />
+                </FormGroup>
+                <FormGroup>
+                  <Label for="select-plan-country">Select a Country</Label>
+                  <Input
+                    id="select-plan-country"
+                    defaultValue={planCountry}
+                    name="select-plan-country"
+                    onChange={onSelectCountryChange}
+                    type="select"
+                  >
+                    <option>...</option>
+                    {/* <option value="BW">Botswana</option> */}
+                    <option value="NA">Namibia</option>
+                    <option value="TH">Thailand</option>
+                  </Input>
+                </FormGroup>
+                <Button
+                  color="primary"
+                  disabled={
+                    // !newPlan.plan_effective_period_end.length ||
+                    // !newPlan.plan_effective_period_start.length ||
+                    !planCountry.length
+                  }
+                  onClick={onStartPlanFormSubmit}
+                >
+                  Select Jurisdictions
+                </Button>
+              </Form>
+            </Col>
+          </Row>
+        </div>
+      );
+    }
+
+    const onEditNameButtonClick = (e: any) => {
```

Comment:
I think we should try very hard not to use `any` here for these events.  Do you agree?

Reply:
I also think that we should consider adding docstrings for these functions here and elsewhere in this module

Reply:
Yeah, I'll go through and add doc strings to all the things today.